pkg/resource/event_bus/custom_update_api.go

```diff
@@ -0,0 +1,146 @@
+package event_bus
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	ackutil "github.com/aws-controllers-k8s/runtime/pkg/util"
+	svcsdk "github.com/aws/aws-sdk-go/service/eventbridge"
+
+	svcapitypes "github.com/aws-controllers-k8s/eventbridge-controller/apis/v1alpha1"
+)
+
+// customUpdateTrail implements a custom logic for handling Trail
+// resource updates.
+func (rm *resourceManager) customUpdateEventBus(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateTrail")
+	defer func(err error) { exit(err) }(err)
```

Comment:
technically, you could just do this instead:
```suggestion
	defer func() { exit(err) }()
```

Reply:
Done, copied verbatim back then from `Trail`.